--- /opt/vllm-ascend/vllm/vllm/model_executor/layers/quantization/__init__.py	2025-05-26 16:39:18.000000000 +0000
+++ __init__.py	2025-05-26 14:25:21.728000000 +0000
@@ -4,6 +4,9 @@
 
 from vllm.model_executor.layers.quantization.base_config import (
     QuantizationConfig)
+from vllm.logger import init_logger
+
+logger = init_logger(__name__)
 
 QuantizationMethods = Literal[
     "aqlm",
@@ -64,8 +67,9 @@
 
     def _wrapper(quant_config_cls):
         if quantization in QUANTIZATION_METHODS:
-            raise ValueError(
-                f"The quantization method `{quantization}` is already exists.")
+            logger.info(f"The ascend adapter based on '{quant_config_cls.__module__.rsplit('.', 1)[-1]}' "
+                        f"quantization is not loaded because an existing adapter is already exists.")
+            return quant_config_cls
         if not issubclass(quant_config_cls, QuantizationConfig):
             raise ValueError("The quantization config must be a subclass of "
                              "`QuantizationConfig`.")
@@ -150,4 +154,4 @@
     "QuantizationMethods",
     "get_quantization_config",
     "QUANTIZATION_METHODS",
-]
\ No newline at end of file
+]
