--- /opt/vllm-ascend/vllm/vllm/config.py	2025-05-26 16:39:18.000000000 +0000
+++ config.py	2025-05-28 11:48:18.000000000 +0000
@@ -772,6 +772,8 @@
             # `override_quantization_method` method) must be checked in order
             # of preference (this is particularly important for GPTQ).
             overrides = [
+                "awq",
+                "compressed-tensors",
                 "marlin",
                 "bitblas",
                 "gptq_marlin_24",
