--- /usr/local/Ascend/atb/atb_llm/models/base/flash_causal_lm.py	2025-05-23 01:57:50.000000000 +0000
+++ flash_causal_lm.py	2025-06-09 11:16:42.836000000 +0000
@@ -54,7 +54,7 @@
         else:
             self.rope_theta = 10000.0
         if hasattr(config, 'rope_scaling') and self.config.rope_scaling is not None:
-            self.scaling_factor = self.config.rope_scaling.factor
+            self.scaling_factor = self.config.rope_scaling["factor"]
         else:
             self.scaling_factor = 1.0
         self.hidden_size = config.hidden_size
