--- /opt/vllm-ascend/vllm/vllm/model_executor/model_loader/loader.py	2025-05-26 16:39:19.000000000 +0000
+++ loader.py	2025-05-28 11:44:27.000000000 +0000
@@ -378,7 +378,6 @@
             else:
                 weights_iterator = safetensors_weights_iterator(
                     hf_weights_files,
-                    self.load_config.use_tqdm_on_load,
                 )
         else:
             weights_iterator = pt_weights_iterator(
