--- model_runner.py	2025-06-10 13:40:05.356000000 +0000
+++ /usr/local/lib/python3.10/dist-packages/vllm_ascend/worker/model_runner.py	2025-05-28 12:27:57.000000000 +0000
@@ -824,7 +824,6 @@
                         second_per_grid_ts=second_per_grid_ts,
                         context_len=inter_data.context_lens[seq_idx],
                         seq_len=inter_data.seq_lens[seq_idx],
-                        audio_feature_lengths=seq_group_metadata.multi_modal_data.get("audio_feature_lengths")
                     )
 
                 seq_data.mrope_position_delta = mrope_position_delta
