--- /usr/local/lib/python3.10/dist-packages/vllm_ascend/platform.py	2025-05-26 16:43:25.000000000 +0000
+++ platform.py	2025-05-28 11:44:45.000000000 +0000
@@ -71,9 +71,6 @@
         from vllm_ascend.utils import adapt_patch
         adapt_patch(is_global_patch=True)
 
-        from vllm_ascend.quantization.quant_config import \
-            AscendQuantConfig  # noqa: F401
-
     @classmethod
     def get_device_capability(cls, device_id: int = 0):
         return None
