--- /opt/vllm-ascend/vllm-ascend/vllm_ascend/__init__.py	2025-04-02 11:37:05.000000000 +0000
+++ /opt/mis/mis/patch/register_patch.py	2025-04-02 11:54:50.470815921 +0000
@@ -18,8 +18,8 @@
 
 def register():
     """Register the NPU platform."""
     # TODO: https://github.com/vllm-project/vllm/pull/12432 Once this pr is merged, 
     # the following module can be imported using pre_register_and_update function.
-    from vllm_ascend.quantization.quant_config import AscendQuantConfig  # noqa: F401
+    # from vllm_ascend.quantization.quant_config import AscendQuantConfig  # noqa: F401
 
     return "vllm_ascend.platform.NPUPlatform"
