--- /opt/vllm-ascend/vllm/vllm/model_executor/layers/quantization/kv_cache.py	2025-05-26 16:39:18.000000000 +0000
+++ kv_cache.py	2025-05-28 01:31:11.000000000 +0000
@@ -51,6 +51,10 @@
         # regardless whether the kv-scale is available in the checkpoint.
         # No need to process kv scales after loading if we are going to
         # calculate them on the fly.
+        if not hasattr(layer, "_q_scale"):
+            layer._q_scale = layer.q_scale
+        if not hasattr(layer, "_prob_scale"):
+            layer._prob_scale = layer.prob_scale
         if layer.kv_cache_dtype != "auto" and not layer.calculate_kv_scales:
             if layer.k_scale > 0.0 and layer.v_scale > 0.0:
                 # We prefer to use separate k_scale and v_scale if present
