--- /usr/local/lib/python3.10/dist-packages/vllm_ascend/worker/model_runner.py	2025-05-28 12:43:57.000000000 +0000
+++ /opt/mis/patch/model_runner.py	2025-06-21 03:24:37.852000000 +0000
@@ -802,9 +802,10 @@
         if self.runner.model_config.uses_mrope:
             image_grid_thw = mm_kwargs.get("image_grid_thw", None)
             video_grid_thw = mm_kwargs.get("video_grid_thw", None)
-            assert image_grid_thw is not None or video_grid_thw is not None, (
+            input_audio_features = mm_kwargs.get("input_audio_features", None)
+            assert image_grid_thw is not None or video_grid_thw is not None or input_audio_features is not None, (
                 "mrope embedding type requires multi-modal input mapper "
-                "returns 'image_grid_thw' or 'video_grid_thw'.")
+                "returns 'image_grid_thw' or 'video_grid_thw' or 'input_audio_features'.")
             second_per_grid_ts = mm_kwargs.get("second_per_grid_ts", None)
 
             hf_config = self.runner.model_config.hf_config
@@ -824,6 +825,7 @@
                         second_per_grid_ts=second_per_grid_ts,
                         context_len=inter_data.context_lens[seq_idx],
                         seq_len=inter_data.seq_lens[seq_idx],
+                        audio_feature_lengths=seq_group_metadata.multi_modal_data.get("audio_feature_lengths"),
                     )
 
                 seq_data.mrope_position_delta = mrope_position_delta
