--- /opt/vllm-ascend/vllm/vllm/model_executor/model_loader/weight_utils.py	2025-05-26 16:39:19.000000000 +0000
+++ weight_utils.py	2025-05-26 09:19:20.368000000 +0000
@@ -429,7 +429,7 @@
 
 def safetensors_weights_iterator(
     hf_weights_files: List[str],
-    use_tqdm_on_load: bool,
+    use_tqdm_on_load: bool = True,
 ) -> Generator[Tuple[str, torch.Tensor], None, None]:
     """Iterate over the weights in the model safetensor files."""
     for st_file in tqdm(
