--- /usr/local/lib/python3.11/dist-packages/vllm/model_executor/layers/quantization/__init__.py	2025-07-29 01:16:52.000000000 +0000
+++ __init__.py	2025-07-29 06:06:09.235804416 +0000
@@ -5,6 +5,9 @@
 
 from vllm.model_executor.layers.quantization.base_config import (
     QuantizationConfig)
+from vllm.logger import init_logger
+
+logger = init_logger(__name__)
 
 QuantizationMethods = Literal[
     "aqlm",
@@ -16,6 +19,7 @@
     "fbgemm_fp8",
     "modelopt",
     "modelopt_fp4",
+    "ms-model-slim",
     "marlin",
     "bitblas",
     "gguf",
@@ -67,8 +71,9 @@
 
     def _wrapper(quant_config_cls):
         if quantization in QUANTIZATION_METHODS:
-            raise ValueError(
-                f"The quantization method `{quantization}` is already exists.")
+            logger.info(f"The ascend adapter based on '{quant_config_cls.__module__.rsplit('.', 1)[-1]}' "
+                        f"quantization is not loaded because an existing adapter is already exists.")
+            return quant_config_cls
         if not issubclass(quant_config_cls, QuantizationConfig):
             raise ValueError("The quantization config must be a subclass of "
                              "`QuantizationConfig`.")
@@ -157,4 +162,4 @@
     "QuantizationMethods",
     "get_quantization_config",
     "QUANTIZATION_METHODS",
-]
\ No newline at end of file
+]
