--- /usr/local/lib/python3.11/dist-packages/vllm_ascend/platform.py	2025-07-29 01:31:59.000000000 +0000
+++ platform.py	2025-07-29 06:06:09.235804416 +0000
@@ -51,7 +51,7 @@
     device_control_env_var: str = "ASCEND_RT_VISIBLE_DEVICES"
     dispatch_key: str = "PrivateUse1"
 
-    supported_quantization: list[str] = [ASCEND_QUATIZATION_METHOD]
+    supported_quantization: list[str] = [ASCEND_QUATIZATION_METHOD, "ms-model-slim"]
 
     def is_sleep_mode_available(self) -> bool:
         return True
@@ -74,9 +74,6 @@
                 if ASCEND_QUATIZATION_METHOD not in quant_action.choices:
                     quant_action.choices.append(ASCEND_QUATIZATION_METHOD)
 
-        from vllm_ascend.quantization.quant_config import \
-            AscendQuantConfig  # noqa: F401
-
     @classmethod
     def get_device_capability(cls, device_id: int = 0):
         return None
